### ISSUE TITLE
[AWAITING PAYMENT] [$250] Chat - Cursor moves before the emoji after adding it by typing on a second line

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48052]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.25-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4895999
**Email or phone of affected tester (no customers):** gatantm@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch new expensify and open any chat
2. Press enter so there is an empty first line
3. On the second line, type :hand: and continue typing after the emoji is typed

## Expected Result:
The cursor goes after the emoji after adding it with its complete code

## Actual Result:
The cursor goes before the just typed emoji, When typing one letter will go before the emoji and the rest after it. It also removes the previous line or if enter is pressed the message is sent instead of adding a new line

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/0ad0bbb5-e21d-4561-98a6-cfdbb064d4ad

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c2f983ff6514acbf</li>
        <li>Upwork Job ID: 1828458402364890340</li>
        <li>Last Price Increase: 2024-08-27</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by garrettmknight** [comment]
Also can't repro on 9.0.25-1. Closing for now.

**Comment by IuliiaHerets** [comment]
The tester added an additional step, issue is still repro on mWeb Chrome Android and Safari iOS.
Steps: 
1. Launch new expensify and open any chat
2. Add a second line of text
3. Type :hand: and continue typing after the emoji is typed

**Comment by garrettmknight** [comment]
@IuliiaHerets I still can't repro on mobile web. Is this an android only bug?

**Comment by NikkiWines** [comment]
I also can't reproduce on iOS

https://github.com/user-attachments/assets/40d0e528-782c-4282-b4ca-06976b0adff9

**Comment by NikkiWines** [comment]
Trying android now

**Comment by IuliiaHerets** [comment]
@garrettmknight @NikkiWines  I can repro on iOS/Safari , so not only Android bug

https://github.com/user-attachments/assets/804fbec3-f96e-4cf9-8b25-44a2e5847dc0

**Comment by NikkiWines** [comment]
Hmm ok, I can't reproduce on dev but I can reproduce on staging mWeb actually - let's see if anything went out on the deploy that would impact this.

**Comment by NikkiWines** [comment]
Also i suspect whatever is causing this is also causing https://github.com/Expensify/App/issues/48051

**Comment by Skalakid** [comment]
@NikkiWines @IuliiaHerets I can't reproduce this issue on the latest staging version `9.0.25-10` iOS/mWeb. There is a chance that other PR just fixed it


https://github.com/user-attachments/assets/7a99dcd9-7a77-4b89-90ce-2ae67299b949

**Comment by mountiny** [comment]
@Skalakid, I can still repro; please note it's crucial you leave the first line empty and press enter. Then on the second line this will fail:

https://github.com/user-attachments/assets/01bff6b4-baac-41ad-b391-d1681097c557

**Comment by Skalakid** [comment]
Okayy, thank you. It's not reproducible when using iOS simulator with connected hardware keyboard. When I use normal iOS keyboard the bug occurs

https://github.com/user-attachments/assets/2a80bf23-915a-4d74-9592-7335a1cbb7cb

**Comment by Skalakid** [comment]
After investigation [my Live Markdown refactor follow-up PR](https://github.com/Expensify/react-native-live-markdown/pull/459) and [one small fix](https://github.com/pulls?q=sort%3Aupdated-desc+is%3Apr+is%3Aopen+author%3A%40me+archived%3Afalse) fixed the problem. Once the second PR is merged, I will create the PR in E/App with the library bump


https://github.com/user-attachments/assets/0766cd8a-f744-4855-b103-2aad13305f95

**Comment by Skalakid** [comment]
Hello, here is a [PR](https://github.com/Expensify/App/pull/48234) with a fix in E/App 😄

**Comment by Skalakid** [comment]
Waiting for the review from @eVoloshchak

**Comment by Skalakid** [comment]
The PR has been merged :D

**Comment by garrettmknight** [comment]
Wow, this one is way late.

@eVoloshchak can you complete the checklist and request payment when you're ready?

**Comment by garrettmknight** [comment]
**Payment Summary:**
- Reviewer: @eVoloshchak $250

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: this wasn't a bug in New Expensify, this was a bug in `react-native-live-markdown` library
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:  N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not needed, we already have the necessary check in the reviewer checklist

**Regression Test Proposal**

1. Open any chat
2. Press enter so there is an empty first line
3. On the second line, enter an emoji (using keyboard, for instance `:hand`) and continue typing after the emoji is typed
4. Verify that the cursor position is correct

**Do we agree 👍 or 👎**

**Comment by JmillsExpensify** [comment]
$250 approved for @eVoloshchak
